### PR TITLE
feat(typography): migrate to @signozhq/ui

### PR DESCRIPTION
**Pull request by H4ad**
## Pull Request

---

Migrate away from Antd Typography by our version that should behave similarly but with less bloat.

Also added a rule to ban usage of Typography from Antd and also ban future components until we remove the package.

| Before | After |
|--------|--------|
| 01-home | 01-home |
| 02-dashboards | 02-dashboards |
|03-dashboard-new | 03-dashboard-new |
| 04-settings | 04-settings |
| 05-settings-general | 05-settings-general |
| 06-settings-org | 06-settings-org |
| 07-settings-members | 07-settings-members |
| 08-settings-service-accounts | 08-settings-service-accounts |
| 09-alerts | 09-alerts |
| 10-alert-type | 10-alert-type |
| 11-logs-explorer | 11-logs-explorer |
| 12-traces-explorer | 12-traces-explorer |
| 13-services | 13-services |
| 14-exceptions |  14-exceptions|
| 15-integrations | 15-integrations |
| 16-metrics-explorer  |  16-metrics-explorer |
| 17-support  | 17-support |
| 18-alerts-config | 18-alerts-config |
| 19-logs-pipelines | 19-logs-pipelines |
| 20-infra-hosts | 20-infra-hosts |
| 21-mq-overview |21-mq-overview  |

---

### ✅ Change Type
_Select all that apply_

- [ ] ✨ Feature
- [ ] 🐛 Bug fix
- [x] ♻️ Refactor
- [ ] 🛠️ Infra / Tooling
- [ ] 🧪 Test-only

---

- Tests added/updated: -
- Manual verification: Yes
- Edge cases covered: -

---

- Blast radius: Entire app
- Potential regressions: Mismatch with previous text or some text not rendering correctly.
- Rollback plan: Fix the issue, no revert.

---

| Field | Value |
|------|-------|
| Deployment Type | Cloud / OSS / Enterprise |
| Change Type | Feature |
| Description | We are slowly transitioning from Antd components to our own SigNoz component library for a lightweight experience. |

---

### 📋 Checklist
- [ ] Tests added or explicitly not required
- [x] Manually tested
- [ ] Breaking changes documented
- [ ] Backward compatibility considered

### CODE REVIEW

File: frontend/src/styles.scss [hunk 1]
```diff
@@ -798,3 +798,12 @@ notifications	-  2050
...
 }
+
+*[data-slot='typography'][data-variant='text'] {
+	display: inline;
+}
+
+*[data-slot='typography'][data-variant='title'] {
+	margin-top: 9px;
+	margin-bottom: 9px;
+}
```
Comment (H4ad) [inline frontend/src/styles.scss:820]: Waiting https://github.com/SigNoz/components/pull/201 to be merged to remove this.

File: frontend/src/pages/UnAuthorized/index.tsx [hunk 2]
```diff
@@ -37,13 +38,15 @@ function UnAuthorizePage(): JSX.Element {
 					It looks like you don&lsquo;t have permission to view this page. <br />
 					{mistakeMessage}
 					{userIsAnonymous ? (
-						<Typography.Link
+						<a
```
Comment (AshwinBhatkal) [inline frontend/src/pages/UnAuthorized/index.tsx]: do we not have a link component for this?

File: frontend/src/pages/MessagingQueues/MQDetails/MQTables/MQTables.tsx [hunk 3]
```diff
@@ -76,7 +77,9 @@ export function getColumns(
 			: (text: string): ColumnTypeRender<Record<string, unknown>> => ({
 					children:
 						column.name === 'service_name' ? (
-							<Typography.Link
+							<a
```
Comment (AshwinBhatkal) [inline frontend/src/pages/MessagingQueues/MQDetails/MQTables/MQTables.tsx]: same here

File: frontend/src/pages/MessagingQueues/MQDetails/DropRateView/DropRateView.tsx [hunk 4]
```diff
@@ -120,15 +125,17 @@ export function getColumns(
 
 			if (column === 'consumer_service' || column === 'producer_service') {
 				return (
-					<Typography.Link
+					<a
```
Comment (AshwinBhatkal) [inline frontend/src/pages/MessagingQueues/MQDetails/DropRateView/DropRateView.tsx]: same

File: frontend/src/container/SpanDetailsDrawer/SpanDetailsDrawer.tsx [hunk 5]
```diff
@@ -610,10 +610,11 @@ function SpanDetailsDrawer(props: ISpanDetailsDrawerProps): JSX.Element {
 													: 'span-percentile-value-container-closed'
 											}`}
 										>
-											<Typography.Text
+											<span
```
Comment (AshwinBhatkal) [inline frontend/src/container/SpanDetailsDrawer/SpanDetailsDrawer.tsx]: why make this span?

File: frontend/src/container/QueryBuilder/components/HavingFilterTag/HavingFilterTag.tsx [hunk 6]
```diff
@@ -13,9 +13,9 @@ export function HavingFilterTag({
 
 	return (
 		<StyledTag closable={closable} onClose={onClose}>
-			<StyledText ellipsis onClick={handleClick}>
-				{value}
-			</StyledText>
+			<span role="button" tabIndex={0} onClick={handleClick}>
```
Comment (AshwinBhatkal) [inline frontend/src/container/QueryBuilder/components/HavingFilterTag/HavingFilterTag.tsx:16]: we have 2 spans here? span within a span

File: frontend/src/container/QueryBuilder/filters/MetricNameSelector/MetricNameSelector.tsx [hunk 7]
```diff
@@ -281,7 +282,7 @@ export const MetricNameSelector = memo(function MetricNameSelector({
 				isFetching ? (
 					<Spin size="small" />
 				) : isError ? (
-					<Typography.Text type="danger" style={{ fontSize: 12 }}>
+					<Typography.Text color="danger" style={{ fontSize: 12 }}>
```
Comment (AshwinBhatkal) [inline frontend/src/container/QueryBuilder/filters/MetricNameSelector/MetricNameSelector.tsx:285]: i feel this attribute should have been type itself and not changed to color. Color could be another attribute. We could avoid style in such cases

File: frontend/src/container/NewWidget/RightContainer/LegendColors/LegendColors.tsx [hunk 8]
```diff
@@ -178,15 +179,17 @@ function LegendColors({
 												</div>
 												{customLegendColors[label] && (
 													<div className="legend-actions">
-														<Typography.Link
+														<a
```
Comment (AshwinBhatkal) [inline frontend/src/container/NewWidget/RightContainer/LegendColors/LegendColors.tsx]: link?

File: frontend/src/container/NoLogs/NoLogs.tsx [hunk 9]
```diff
@@ -62,9 +62,14 @@ export default function NoLogs({
 					</span>
 				</Typography>
 
-				<Typography.Link className="send-logs-link" onClick={handleLinkClick}>
+				<a
```
Comment (AshwinBhatkal) [inline frontend/src/container/NoLogs/NoLogs.tsx]: link

File: frontend/src/container/ListOfDashboard/TableComponents/styles.ts [hunk 10]
```diff
@@ -1,7 +1,8 @@
-import { Typography } from 'antd';
 import styled from 'styled-components';
 
-export const TableLinkText = styled(Typography.Text)`
-	color: #4e74f8 !important;
-	cursor: pointer;
+export const TableLinkText = styled.span<{ disabled: boolean }>`
```
Comment (AshwinBhatkal) [inline frontend/src/container/ListOfDashboard/TableComponents/styles.ts:3]: why not keep typo.text here?

File: frontend/src/container/Login/index.tsx [hunk 11]
```diff
@@ -359,7 +360,9 @@ function Login(): JSX.Element {
 						<ParentContainer>
 							<div className="password-label-container">
 								<Label htmlFor="Password">Password</Label>
-								<Typography.Link
+								<a
```
Comment (AshwinBhatkal) [inline frontend/src/container/Login/index.tsx]: link

File: frontend/src/container/MetricsApplication/Tabs/Overview/TableRenderer/ColumnWithLink.tsx [hunk 12]
```diff
@@ -49,11 +49,13 @@ function ColumnWithLink({
 
 	return (
 		<Tooltip placement="topLeft" title={text}>
-			<Typography.Link
+			<a
```
Comment (AshwinBhatkal) [inline frontend/src/container/MetricsApplication/Tabs/Overview/TableRenderer/ColumnWithLink.tsx]: link

File: frontend/src/container/MetricsApplication/TopOperationsTable.tsx [hunk 13]
```diff
@@ -117,7 +116,9 @@ function TopOperationsTable({
 		},
 		render: (text: string): JSX.Element => (
 			<Tooltip placement="topLeft" title={text}>
-				<Typography.Link
+				<a
```
Comment (AshwinBhatkal) [inline frontend/src/container/MetricsApplication/TopOperationsTable.tsx]: link

File: frontend/src/container/MetricsExplorer/Explorer/TimeSeries.tsx [hunk 14]
```diff
@@ -287,11 +288,13 @@ function TimeSeries({
 										title={
 											<Typography.Text>
 												No unit is set for this metric. You can assign one from the{' '}
-												<Typography.Link
+												<a
```
Comment (AshwinBhatkal) [inline frontend/src/container/MetricsExplorer/Explorer/TimeSeries.tsx]: link

File: frontend/src/container/CreateAlertRule/SelectAlertType/index.tsx [hunk 15]
```diff
@@ -78,15 +79,17 @@ function SelectAlertType({ onSelect }: SelectAlertTypeProps): JSX.Element {
 						data-testid={`alert-type-card-${option.selection}`}
 					>
 						{option.description}{' '}
-						<Typography.Link
+						<a
```
Comment (AshwinBhatkal) [inline frontend/src/container/CreateAlertRule/SelectAlertType/index.tsx]: link

File: frontend/src/container/BillingContainer/BillingContainer.tsx [hunk 16]
```diff
@@ -362,15 +362,19 @@ export default function BillingContainer(): JSX.Element {
 		[apiResponse, billAmount, isLoading, isFetchingBillingData],
 	);
 
-	const { Text } = Typography;
 	const subscriptionPastDueMessage = (): JSX.Element => (
 		<Typography>
 			{`We were not able to process payments for your account. Please update your card details `}
-			<Text type="danger" onClick={handleBilling} style={{ cursor: 'pointer' }}>
+			<a
```
Comment (AshwinBhatkal) [inline frontend/src/container/BillingContainer/BillingContainer.tsx]: link

File: frontend/src/pages/UnAuthorized/index.tsx (hunk 2 above)
Comment (H4ad) [reply frontend/src/pages/UnAuthorized/index.tsx]: I didn't add support for onClick on Typography.Link

File: frontend/src/pages/MessagingQueues/MQDetails/MQTables/MQTables.tsx (hunk 3 above)
Comment (H4ad) [reply frontend/src/pages/MessagingQueues/MQDetails/MQTables/MQTables.tsx]: I didn't add support for onClick on Typography.Link

File: frontend/src/pages/MessagingQueues/MQDetails/DropRateView/DropRateView.tsx (hunk 4 above)
Comment (H4ad) [reply frontend/src/pages/MessagingQueues/MQDetails/DropRateView/DropRateView.tsx]: I didn't add support for onClick on Typography.Link

File: frontend/src/container/SpanDetailsDrawer/SpanDetailsDrawer.tsx (hunk 5 above)
Comment (H4ad) [reply frontend/src/container/SpanDetailsDrawer/SpanDetailsDrawer.tsx]: Because of the onClick

File: frontend/src/container/QueryBuilder/components/HavingFilterTag/HavingFilterTag.tsx (hunk 6 above)
Comment (H4ad) [reply frontend/src/container/QueryBuilder/components/HavingFilterTag/HavingFilterTag.tsx:16]: onClick again

File: frontend/src/container/NewWidget/RightContainer/LegendColors/LegendColors.tsx (hunk 8 above)
Comment (H4ad) [reply frontend/src/container/NewWidget/RightContainer/LegendColors/LegendColors.tsx]: onClick support

File: frontend/src/container/NoLogs/NoLogs.tsx (hunk 9 above)
Comment (H4ad) [reply frontend/src/container/NoLogs/NoLogs.tsx]: onClick support

File: frontend/src/container/ListOfDashboard/TableComponents/styles.ts (hunk 10 above)
Comment (H4ad) [reply frontend/src/container/ListOfDashboard/TableComponents/styles.ts:3]: disabled property support and onClick

File: frontend/src/container/QueryBuilder/filters/MetricNameSelector/MetricNameSelector.tsx (hunk 7 above)
Comment (H4ad) [reply frontend/src/container/QueryBuilder/filters/MetricNameSelector/MetricNameSelector.tsx:285]: I don't agree with type, does not make much sense to change the color of a text element with type.

**Review by makeavish — DISMISSED** [review]

**Review by AshwinBhatkal — DISMISSED** [review]

**Review by makeavish — DISMISSED** [review]

**Review by AshwinBhatkal — DISMISSED** [review]

**Review by YounixM — DISMISSED** [review]

**Review by makeavish — APPROVED** [review]

**Review by YounixM — APPROVED** [review]

**Review by ahrefabhi — APPROVED** [review]
